packages/runtime/datastore/src/localChannelContext.ts

```diff
@@ -156,12 +156,15 @@ export class RehydratedLocalChannelContext extends LocalChannelContextBase {
         private readonly snapshotTree: ISnapshotTree,
     ) {
         super(id, registry, runtime, () => this.services);
-        let blobMap: Map<string, ArrayBufferLike> | undefined;
+        const blobMap: Map<string, ArrayBufferLike> = new Map<string, ArrayBufferLike>();
         const clonedSnapshotTree = cloneDeep(this.snapshotTree);
-        if (clonedSnapshotTree !== undefined) {
-            blobMap = new Map<string, ArrayBufferLike>();
-            this.collectExtraBlobsAndSanitizeSnapshot(clonedSnapshotTree, blobMap);
+        // 0.47 back-compat Need to sanitize if snapshotTree.blobs still contains blob contents too.
+        // This is for older snapshot which still contains the contents within blobs.
+        // After a couple of revisions we can remove it.
+        if (this.checkNeedToSanitizeAndCollectBlobs(clonedSnapshotTree, blobMap)) {
```

Comment:
This is for serialized containers created in past versions, right? If so, might be worth refining comment to make it more clear

Reply:
Where old loaders are used.